### PR TITLE
cppcheck: fix some defects found by new cppcheck

### PR DESCRIPTION
**Pull request by disaykin**
* The one definition rule is violated, different classes/structs have the same name 'injector'
 * src/plugins/procdump/minidump.h and src/plugins/procdump2/minidump2.h code duplication
 * src/plugins/procdump/writer.cpp and src/plugins/procdump2/writer.cpp code duplication

### CODE REVIEW

**Comment by tklengyel** [comment]
You can also switch back the github actions runner to be `ubuntu-latest` with this PR

**Comment by disaykin** [comment]
I fix other errors and switch back the cppcheck github action to be `ubuntu-latest`

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

File: src/libinjector/injector.c [hunk 1]
```diff
@@ -116,7 +118,7 @@ injector_status_t injector_start_app(
     const char* binary_path,
     const char* target_process,
     bool break_loop_on_detection,
-    injector_t* injector_to_be_freed,
+    void** injector_to_be_freed,
```
Comment (tklengyel) [inline src/libinjector/injector.c:121]: Switching this to `void**` I don't think is helping with readability of the code.

File: src/plugins/Makefile.am [hunk 2]
```diff
@@ -280,9 +280,9 @@ if PLUGIN_PROCDUMP2
 sources += procdump2/procdump2.cpp
 sources += procdump2/procdump2.h
 sources += procdump2/private2.h
-sources += procdump2/writer.cpp
-sources += procdump2/writer.h
-sources += procdump2/minidump2.h
+sources += procdump/writer.cpp
+sources += procdump/writer.h
+sources += procdump/minidump.h
```
Comment (tklengyel) [inline src/plugins/Makefile.am:285]: This doesn't look right. One plugin should never include files/headers from another one. In this case it's pretty clear one would only be using one version of the plugin so whatever cppcheck is complaining about is likely due to it trying with both being enabled. Adding an assert to ensure that never happens would be the right solution.

File: src/libinjector/injector.c (hunk 1 above)
Comment (disaykin) [reply src/libinjector/injector.c:121]: I agree. But I don't know how to fix this properly. Client code does not know, `linux_injector_t` or `injector_t` will return. This hack needed only for windows use case now.

File: src/plugins/Makefile.am (hunk 2 above)
Comment (disaykin) [reply src/plugins/Makefile.am]: I fix it using namespaces

File: src/libinjector/injector.c (hunk 1 above)
Comment (disaykin) [reply src/libinjector/injector.c]: Done

**Comment by tklengyel** [comment]
@drakvuf-jenkins This is OK to test

**Pull request merged**